sbi/inference/base.py

```diff
@@ -173,29 +185,24 @@ def provide_presimulated(
             from_round: Which round the data was simulated from. `from_round=0` means
                 that the data came from the first round, i.e. the prior.
         """
-        self._append_to_data_bank(theta, x, from_round)
-
-    def _append_to_data_bank(self, theta: Tensor, x: Tensor, from_round: int) -> None:
-        r"""
-        Store data in as entries in a list for each type of variable (parameter/data).
-
-        Stores $\theta$, $x$, prior_masks (indicating if simulations are coming from the
-        prior or not) and a index indicating which round the batch of simulations came
-        from.
-
-        Args:
-            theta: Parameter sets.
-            x: Simulated data.
-            from_round: What round the $(\theta, x)$ pairs are coming from. We start
-                counting from round 0.
-        """
+        raise NameError(
+            ".provide_presimulated() does no longer exist in sbi "
+            "versions >=0.14.0. Instead, simply pass theta and x to "
+            ".__call__()."
+            "Please consult "
+            "the corresponding pull request on github: "
+            "https://github.com/mackelab/sbi/pull/378 and tutorials: "
+            "https://www.mackelab.org/sbi/tutorial/02_flexible_interface/ for further "
+            "information."
+        )
 
-        self._theta_roundwise.append(theta)
-        self._x_roundwise.append(x)
-        self._prior_masks.append(mask_sims_from_prior(from_round, theta.size(0)))
-        self._data_round_index.append(from_round)
+    @abstractmethod
+    def add_data(
+        self, theta: Tensor, x: Tensor, theta_from_proposal: Union[bool, int] = False
+    ) -> "NeuralInference":
+        raise NotImplementedError
```

Comment:
Why can't we make this method agnostic to the inference method? 
also the signature here seems to be different from that in `snle_base`, no?

Reply:
See comment above.

But I'll remove this from `base.py` since, indeed, the signature in `SNLE` is different.